lib/read/test/Cardano/Wallet/Read/Tx/TxIdSpec.hs

```diff
@@ -0,0 +1,255 @@
+    \00005840e8e769ecd0f3c538f0a5a574a1c881775f086d6f4c845b\
+    \81be9b78955728bffa7efa54297c6a5d73337bd6280205b1759c13\
+    \f79d4c93f29871fc51b78aeba80e58200000000000000000000000\
+    \00000000000000000000000000000000000000000044a1024100f6"
+
+alonzoTxId :: TxId
+alonzoTxId = unsafeTxIdFromHex
+    "ca011f22d07b97ee17f6f2e2ef568b9521791608169425e92993c8c6e5541d79"
+
+alonzoTx :: Tx Read.Alonzo
+alonzoTx = unsafeParseEraTxFromHex
+    "84a400828258200000000000000000000000000000000000000000\
+    \000000000000000000000000008258200000000000000000000000\
+    \000000000000000000000000000000000000000000010183825839\
+    \010202020202020202020202020202020202020202020202020202\
+    \020202020202020202020202020202020202020202020202020202\
+    \0202021a005b8d8082583901030303030303030303030303030303\
+    \030303030303030303030303030303030303030303030303030303\
+    \03030303030303030303030303031a005b8d808258390104040404\
+    \040404040404040404040404040404040404040404040404040404\
+    \040404040404040404040404040404040404040404040404041a00\
+    \7801e0021a0002102003191e46a10282845820130ae82201d7072e\
+    \6fbfc0a1884fb54636554d14945b799125cf7ce38d477f51584058\
+    \35ff78c6fc5e4466a179ca659fa85c99b8a3fba083f3f3f42ba360\
+    \d479c64ef169914b52ade49b19a7208fd63a6e67a19c406b482660\
+    \8fdc5307025506c307582001010101010101010101010101010101\
+    \0101010101010101010101010101010144a1024100845820010000\
+    \000000000000000000000000000000000000000000000000000000\
+    \00005840e8e769ecd0f3c538f0a5a574a1c881775f086d6f4c845b\
+    \81be9b78955728bffa7efa54297c6a5d73337bd6280205b1759c13\
+    \f79d4c93f29871fc51b78aeba80e58200000000000000000000000\
+    \00000000000000000000000000000000000000000044a1024100f5\
+    \f6"
+
+babbageTxId :: TxId
+babbageTxId = unsafeTxIdFromHex
+    "1b81e33957d8b96e17a142ae06004213d25dc6abed0fc3949a16f3a96eb2a093"
+
+babbageTx :: Tx Read.Babbage
+babbageTx = unsafeParseEraTxFromHex
+    "84a400818258200000000000000000000000000000000000000000\
+    \000000000000000000000000000182a20058390101010101010101\
+    \010101010101010101010101010101010101010101010101010101\
+    \01010101010101010101010101010101010101010101011a001e84\
+    \80a200583901020202020202020202020202020202020202020202\
+    \020202020202020202020202020202020202020202020202020202\
+    \0202020202020202011a0078175c021a0001faa403191e46a10281\
+    \845820010000000000000000000000000000000000000000000000\
+    \000000000000000058407154db81463825f150bb3b9b0824caf151\
+    \3716f73498afe61d917a5621912a2b3df252bea14683a9ee56710d\
+    \483a53a5aa35247e0d2b80e6300f7bdec763a20458200000000000\
+    \000000000000000000000000000000000000000000000000000000\
+    \44a1024100f5f6"
+
+unsafeParseEraTxFromHex
+    :: forall era. IsEra era
+    => ByteString
+    -> Tx era
+unsafeParseEraTxFromHex bytes =
+    unsafeFromEraValue
+    . either (error . show) id
+    . parseTxFromCBOR
+    $ EraValue (K $ unsafeReadBase16 bytes :: K BL.ByteString era)
```

Comment:
🧐 `fromEraValue` seems sketchy to me. Should it really exist? What if we here instead wrote:

```suggestion
    either (error . show) id
    $ unComp
    $ deserializeTx
    (K $ unsafeReadBase16 bytes :: K BL.ByteString era)
  where
    unComp :: (f :.: g) era -> f (g era)
    unComp (Comp fg) = fg
```

Reply:
The function `fromEraValue` is no more fishy than the function `read :: Read a => String → a` — this function is not parametrically polymorphic, the result value depends heavily on the result type.

A better name for the function `fromEraValue` would probably be `matchEraValue`, though.

But sure, I can remove this function.

Reply:
> A better name for the function fromEraValue would probably be matchEraValue, though.

👍